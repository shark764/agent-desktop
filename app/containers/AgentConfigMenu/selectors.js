import { createSelector } from 'reselect';

/**
* Other specific selectors
*/

const selectToolbarDomain = (state) => state.get('toolbar');

const selectEnabledStats = createSelector(
  selectToolbarDomain,
  (toolbar) => toolbar.get('enabledStats').toJS()
);

const selectAvailableStats = createSelector(
  selectToolbarDomain,
  (toolbar) => toolbar.get('availableStats').toJS()
);

export {
  selectEnabledStats,
  selectAvailableStats,
};
