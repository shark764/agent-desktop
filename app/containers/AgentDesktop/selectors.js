/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

import { createSelector } from 'reselect';

const selectLoginMap = (state) => state.get('login');
const selectAgentDesktopMap = (state) => state.get('agentDesktop');

const selectAgentId = createSelector(
  selectLoginMap,
  (login) => login.get('agent').get('userId')
);

const selectIsAgentReady = createSelector(
  selectAgentDesktopMap,
  (agentDesktop) => agentDesktop.get('presence') === 'ready'
);

const selectInteractions = createSelector(
  selectAgentDesktopMap,
  (agentDesktop) => agentDesktop.get('interactions')
);

const getSelectedInteractionId = createSelector(
  selectAgentDesktopMap,
  (agentDesktop) => agentDesktop.get('selectedInteractionId')
);

const selectNewInteractionPanel = createSelector(
  selectAgentDesktopMap,
  (agentDesktop) => agentDesktop.get('newInteractionPanel')
);
const selectQueues = createSelector(
  selectAgentDesktopMap,
  (agentDesktop) => agentDesktop.get('queues').toJS()
);

const selectSelectedInteraction = createSelector(
  [selectNewInteractionPanel, selectInteractions, getSelectedInteractionId],
  (newInteractionPanel, interactions, selectedInteractionId) => {
    if (selectedInteractionId !== undefined) {
      if (selectedInteractionId === 'creating-new-interaction') {
        return newInteractionPanel.toJS();
      } else {
        return interactions.toJS().find((interaction) =>
          interaction.interactionId === selectedInteractionId
        );
      }
    } else {
      return undefined;
    }
  }
 );

const selectAwaitingDisposition = createSelector(
  selectSelectedInteraction,
  (interaction) => (
    typeof interaction !== 'undefined' &&
    interaction.status === 'wrapup' &&
    interaction.dispositionDetails.forceSelect &&
    interaction.dispositionDetails.selected.length === 0
  )
);

const selectHasVoiceInteraction = createSelector(
  selectInteractions,
  (interactions) => interactions.findIndex(
    (interaction) => (interaction.get('channelType') === 'voice')
  ) !== -1
);

const selectSmsInteractionNumbers = createSelector(
  selectInteractions,
  (interactions) => {
    const smsInteractionNumbers = [];
    interactions.forEach((interaction) => {
      if (interaction.get('channelType') === 'sms') {
        smsInteractionNumbers.push(interaction.get('customer'));
      }
    });
    return smsInteractionNumbers;
  }
);

const selectIsContactsPanelCollapsed = createSelector(
  selectAgentDesktopMap,
  (agentDesktop) => agentDesktop.get('isContactsPanelCollapsed')
);

export {
  selectAgentId,
  selectIsAgentReady,
  selectInteractions,
  getSelectedInteractionId,
  selectLoginMap,
  selectAgentDesktopMap,
  selectSelectedInteraction,
  selectAwaitingDisposition,
  selectHasVoiceInteraction,
  selectSmsInteractionNumbers,
  selectQueues,
  selectIsContactsPanelCollapsed,
};
