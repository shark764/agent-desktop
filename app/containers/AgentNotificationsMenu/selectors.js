/*
 * Copyright Â© 2015-2017 Serenova, LLC. All rights reserved.
 */

const selectNotificationPreferences = (state) =>
  state.get('notificationPreferences');

export const selectAudioPreferences = (state) =>
  selectNotificationPreferences(state).get('audioNotifications');

export const selectVisualPreferences = (state) =>
  Notification.permission === 'granted' &&
  selectNotificationPreferences(state).get('visualNotifications');
