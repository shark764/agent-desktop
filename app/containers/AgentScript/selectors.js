import { createSelector } from 'reselect';

/**
 * Direct selector to the agentScript state domain
 */
const selectAgentScriptDomain = () => (state) => state.get('agentScript');

/**
 * Other specific selectors
 */


/**
 * Default selector used by AgentScript
 */

const selectAgentScript = () => createSelector(
  selectAgentScriptDomain(),
  (substate) => substate.toJS()
);

export default selectAgentScript;
export {
  selectAgentScriptDomain,
};
