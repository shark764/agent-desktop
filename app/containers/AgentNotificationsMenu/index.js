import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { FormattedMessage } from 'react-intl';
import Radium from 'radium';
import { isIeEleven } from 'utils/browser';

import Icon from 'components/Icon';

import { selectAudioPreferences, selectVisualPreferences } from './selectors';
import {
  toggleAudioNotificationsPreference,
  toggleVisualNotificationsPreference,
} from './actions';
import messages from './messages';

const styles = {
  notificationOption: {
    padding: '5px',
    cursor: 'pointer',
    ':hover': {
      backgroundColor: '#DEF8FE',
    },
  },
  checkStatus: {
    float: 'right',
  },
};

export function AgentNotificationsMenu(props) {
  return (
    <Fragment>
      <div
        id="audioNotificationOption"
        key="audioNotificationOption"
        onClick={props.toggleAudioNotificationsPreference}
        style={styles.notificationOption}
      >
        <FormattedMessage {...messages.audio} />
        {props.audioNotificationsEnabled && (
          <Icon name="checkStatus" style={styles.checkStatus} />
        )}
      </div>
      {!isIeEleven() &&
        window.parent === window && (
        <div
          id="visualNotificationOption"
          key="visualNotificationOption"
          onClick={props.toggleVisualNotificationsPreference}
          style={styles.notificationOption}
        >
          <FormattedMessage {...messages.visual} />
          {props.visualNotificationsEnabled && (
            <Icon name="checkStatus" style={styles.checkStatus} />
          )}
        </div>
      )}
    </Fragment>
  );
}

function mapStateToProps(state, props) {
  return {
    audioNotificationsEnabled: selectAudioPreferences(state, props),
    visualNotificationsEnabled: selectVisualPreferences(state, props),
  };
}

const actions = {
  toggleAudioNotificationsPreference,
  toggleVisualNotificationsPreference,
};

AgentNotificationsMenu.propTypes = {
  audioNotificationsEnabled: PropTypes.bool,
  visualNotificationsEnabled: PropTypes.bool,
  toggleAudioNotificationsPreference: PropTypes.func.isRequired,
  toggleVisualNotificationsPreference: PropTypes.func.isRequired,
};

export default connect(
  mapStateToProps,
  actions
)(Radium(AgentNotificationsMenu));
