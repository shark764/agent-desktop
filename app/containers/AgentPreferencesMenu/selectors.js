/*
 * Copyright Â© 2015-2018 Serenova, LLC. All rights reserved.
 */

import { createSelector } from 'reselect';

import { getCurrentTenantPermissions } from 'containers/App/selectors';

export const selectHasViewStatsPermission = createSelector(
  [getCurrentTenantPermissions],
  (tenantPermissions) =>
    tenantPermissions.includes('AGENT_EXPERIENCE_METRICS_VIEW')
);
