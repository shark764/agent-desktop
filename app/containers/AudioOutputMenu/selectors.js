/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

import { createSelector } from '@reduxjs/toolkit';

import { selectActiveExtension } from 'containers/AgentStatusMenu/selectors';

const selectAudioOutputPreferences = (state) =>
  state.get('audioOutputPreferences');

/**
 * Available connected devices
 * Returns array containing:
 * {
 *    id: "default"
 *    label: "Default"
 * }
 */
export const getAvailableOutputDevices = (state) =>
  selectAudioOutputPreferences(state).availableOutputDevices;

/**
 * Active connected ringtone devices
 * Returns array of ids
 */

export const getActiveOutputRingtoneDevices = (state) =>
  selectAudioOutputPreferences(state).activeOutputRingtoneDevices;

export const selectActiveOutputRingtoneDevices = createSelector(
  [getAvailableOutputDevices, getActiveOutputRingtoneDevices],
  (availables, ringtones) =>
    availables.map((device) => ({
      ...device,
      isActive: ringtones.includes(device.id),
    }))
);

/**
 * Active connected speaker devices
 * Returns array of ids
 */

export const getActiveOutputSpeakerDevices = (state) =>
  selectAudioOutputPreferences(state).activeOutputSpeakerDevices;

export const selectActiveOutputSpeakerDevices = createSelector(
  [getAvailableOutputDevices, getActiveOutputSpeakerDevices],
  (availables, speakers) =>
    availables.map((device) => ({
      ...device,
      isActive: speakers.includes(device.id),
    }))
);

/**
 * Active connected notification devices
 * Returns array of ids
 */

export const getActiveOutputNotificationDevices = (state) =>
  selectAudioOutputPreferences(state).activeOutputNotificationDevices;

export const selectActiveOutputNotificationDevices = createSelector(
  [getAvailableOutputDevices, getActiveOutputNotificationDevices],
  (availables, notifications) =>
    availables.map((device) => ({
      ...device,
      isActive: notifications.includes(device.id),
    }))
);

/**
 * Returns if choosing devices is supported by browser
 */
export const selectOutputSelectionSupported = (state) =>
  selectAudioOutputPreferences(state).isOutputSelectionSupported;

/**
 * Returns if active extension is Twilio/WebRTC
 */
export const selectActiveExtensionIsTwilio = createSelector(
  selectActiveExtension,
  (activeExtension) =>
    activeExtension.provider === 'twilio' && activeExtension.type === 'webrtc'
);
