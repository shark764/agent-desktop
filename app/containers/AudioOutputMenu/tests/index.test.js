/*
 * Copyright Â© 2015-2020 Serenova, LLC. All rights reserved.
 */

import React from 'react';
import { createStore } from 'redux';
import { shallow } from 'enzyme';

import { isToolbar } from 'utils/url';
import { selectCrmModule } from 'containers/AgentDesktop/selectors';
import { selectAudioPreferences } from 'containers/AgentNotificationsMenu/selectors';

import {
  selectActiveOutputRingtoneDevices,
  selectActiveOutputSpeakerDevices,
  selectActiveOutputNotificationDevices,
} from '../selectors';

import ConnectedAudioOutputMenu, { AudioOutputMenu } from '..';

jest.mock('utils/url');
isToolbar.mockImplementation(() => false);

const ringtoneDevices = [
  { id: 'default', label: 'Default', isActive: true },
  {
    id: 'aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b',
    label: 'Built-in Audio Analog Stereo',
    isActive: true,
  },
  {
    id: '6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb',
    label: 'Plantronics C320 Analog Stereo',
    isActive: false,
  },
  {
    id: '807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957',
    label: 'USG PnIXAu20o  Analog Stereo',
    isActive: true,
  },
];
const updateActiveOutputRingtoneDevice = jest.fn();
const speakerDevices = [
  { id: 'default', label: 'Default', isActive: false },
  {
    id: 'aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b',
    label: 'Built-in Audio Analog Stereo',
    isActive: true,
  },
  {
    id: '6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb',
    label: 'Plantronics C320 Analog Stereo',
    isActive: true,
  },
  {
    id: '807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957',
    label: 'USG PnIXAu20o  Analog Stereo',
    isActive: false,
  },
];
const updateActiveOutputSpeakerDevice = jest.fn();
const notificationDevices = [
  { id: 'default', label: 'Default', isActive: false },
  {
    id: 'aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b',
    label: 'Built-in Audio Analog Stereo',
    isActive: false,
  },
  {
    id: '6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb',
    label: 'Plantronics C320 Analog Stereo',
    isActive: false,
  },
  {
    id: '807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957',
    label: 'USG PnIXAu20o  Analog Stereo',
    isActive: true,
  },
];
const updateActiveOutputNotificationDevice = jest.fn();

jest.mock('containers/AgentDesktop/selectors');
selectCrmModule.mockImplementation(() => 'none');
jest.mock('containers/AgentNotificationsMenu/selectors');
selectAudioPreferences.mockImplementation(() => true);
jest.mock('../selectors');
selectActiveOutputRingtoneDevices.mockImplementation(() => ringtoneDevices);
selectActiveOutputSpeakerDevices.mockImplementation(() => speakerDevices);
selectActiveOutputNotificationDevices.mockImplementation(
  () => notificationDevices
);

describe('<AudioOutputMenu />', () => {
  const store = createStore((state) => state);
  const testProps = {
    store,
    ringtoneDevices,
    speakerDevices,
    notificationDevices,
    updateActiveOutputRingtoneDevice,
    updateActiveOutputSpeakerDevice,
    updateActiveOutputNotificationDevice,
    audioNotificationsEnabled: true,
    crmModule: 'none',
  };

  it('renders the audio output menu', () => {
    const rendered = shallow(<AudioOutputMenu {...testProps} />);
    expect(rendered).toMatchSnapshot();
  });

  it('renders the audio output menu when devices lists are empty', () => {
    testProps.ringtoneDevices = [];
    testProps.speakerDevices = [];
    testProps.notificationDevices = [];
    const rendered = shallow(<AudioOutputMenu {...testProps} />);
    expect(rendered).toMatchSnapshot();
  });

  it('renders the audio output menu when notifications are not enabled', () => {
    testProps.audioNotificationsEnabled = false;
    const rendered = shallow(<AudioOutputMenu {...testProps} />);
    expect(rendered).toMatchSnapshot();
  });

  it('renders the audio output menu when crmModule is zendesk in desktop mode', () => {
    testProps.crmModule = 'zendesk';
    const rendered = shallow(<AudioOutputMenu {...testProps} />);
    expect(rendered).toMatchSnapshot();
  });

  it('renders the audio output menu when crmModule is zendesk in toolbar mode', () => {
    isToolbar.mockImplementationOnce(() => true);
    testProps.crmModule = 'zendesk';
    const rendered = shallow(<AudioOutputMenu {...testProps} />);
    expect(rendered).toMatchSnapshot();
  });

  it('renders the audio output menu when crmModule is not zendesk and notifications are not enabled in toolbar mode', () => {
    isToolbar.mockImplementationOnce(() => true);
    testProps.crmModule = 'none';
    testProps.audioNotificationsEnabled = false;
    const rendered = shallow(<AudioOutputMenu {...testProps} />);
    expect(rendered).toMatchSnapshot();
  });

  it('renders the audio output menu connected with redux', () => {
    const rendered = shallow(<ConnectedAudioOutputMenu {...testProps} />);
    expect(rendered).toMatchSnapshot();
  });

  it('renders the audio output menu connected with redux when notifications are not available', () => {
    selectAudioPreferences.mockImplementationOnce(() => false);
    const rendered = shallow(<ConnectedAudioOutputMenu {...testProps} />);
    expect(rendered).toMatchSnapshot();
  });

  it('renders the audio output menu connected with redux when notifications are not available in zendesk mode', () => {
    isToolbar.mockImplementationOnce(() => true);
    selectCrmModule.mockImplementationOnce(() => 'zendesk');
    selectAudioPreferences.mockImplementationOnce(() => false);
    const rendered = shallow(<ConnectedAudioOutputMenu {...testProps} />);
    expect(rendered).toMatchSnapshot();
  });
});
