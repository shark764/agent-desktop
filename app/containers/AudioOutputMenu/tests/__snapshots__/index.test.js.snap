// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioOutputMenu /> renders the audio output menu 1`] = `
<React.Fragment>
  <DeviceMenu
    audio="media"
    devices={
      Array [
        Object {
          "id": "default",
          "isActive": false,
          "label": "Default",
        },
        Object {
          "id": "aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b",
          "isActive": false,
          "label": "Built-in Audio Analog Stereo",
        },
        Object {
          "id": "6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb",
          "isActive": false,
          "label": "Plantronics C320 Analog Stereo",
        },
        Object {
          "id": "807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957",
          "isActive": true,
          "label": "USG PnIXAu20o  Analog Stereo",
        },
      ]
    }
    label={
      Object {
        "defaultMessage": "Media",
        "id": "app.containers.AudioOutputMenu.media",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
  <DeviceMenu
    audio="ringtone"
    devices={
      Array [
        Object {
          "id": "default",
          "isActive": true,
          "label": "Default",
        },
        Object {
          "id": "aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b",
          "isActive": true,
          "label": "Built-in Audio Analog Stereo",
        },
        Object {
          "id": "6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb",
          "isActive": false,
          "label": "Plantronics C320 Analog Stereo",
        },
        Object {
          "id": "807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957",
          "isActive": true,
          "label": "USG PnIXAu20o  Analog Stereo",
        },
      ]
    }
    label={
      Object {
        "defaultMessage": "Ringtone",
        "id": "app.containers.AudioOutputMenu.ringtone",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
  <DeviceMenu
    audio="voice"
    devices={
      Array [
        Object {
          "id": "default",
          "isActive": false,
          "label": "Default",
        },
        Object {
          "id": "aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b",
          "isActive": true,
          "label": "Built-in Audio Analog Stereo",
        },
        Object {
          "id": "6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb",
          "isActive": true,
          "label": "Plantronics C320 Analog Stereo",
        },
        Object {
          "id": "807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957",
          "isActive": false,
          "label": "USG PnIXAu20o  Analog Stereo",
        },
      ]
    }
    label={
      Object {
        "defaultMessage": "Voice",
        "id": "app.containers.AudioOutputMenu.voice",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
</React.Fragment>
`;

exports[`<AudioOutputMenu /> renders the audio output menu connected with redux 1`] = `
<AudioOutputMenu
  audioNotificationsEnabled={true}
  crmModule="none"
  notificationDevices={
    Array [
      Object {
        "id": "default",
        "isActive": false,
        "label": "Default",
      },
      Object {
        "id": "aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b",
        "isActive": false,
        "label": "Built-in Audio Analog Stereo",
      },
      Object {
        "id": "6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb",
        "isActive": false,
        "label": "Plantronics C320 Analog Stereo",
      },
      Object {
        "id": "807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957",
        "isActive": true,
        "label": "USG PnIXAu20o  Analog Stereo",
      },
    ]
  }
  ringtoneDevices={
    Array [
      Object {
        "id": "default",
        "isActive": true,
        "label": "Default",
      },
      Object {
        "id": "aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b",
        "isActive": true,
        "label": "Built-in Audio Analog Stereo",
      },
      Object {
        "id": "6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb",
        "isActive": false,
        "label": "Plantronics C320 Analog Stereo",
      },
      Object {
        "id": "807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957",
        "isActive": true,
        "label": "USG PnIXAu20o  Analog Stereo",
      },
    ]
  }
  speakerDevices={
    Array [
      Object {
        "id": "default",
        "isActive": false,
        "label": "Default",
      },
      Object {
        "id": "aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b",
        "isActive": true,
        "label": "Built-in Audio Analog Stereo",
      },
      Object {
        "id": "6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb",
        "isActive": true,
        "label": "Plantronics C320 Analog Stereo",
      },
      Object {
        "id": "807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957",
        "isActive": false,
        "label": "USG PnIXAu20o  Analog Stereo",
      },
    ]
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
  updateActiveOutputNotificationDevice={[Function]}
  updateActiveOutputRingtoneDevice={[Function]}
  updateActiveOutputSpeakerDevice={[Function]}
/>
`;

exports[`<AudioOutputMenu /> renders the audio output menu connected with redux when notifications are not available 1`] = `
<AudioOutputMenu
  audioNotificationsEnabled={false}
  crmModule="none"
  notificationDevices={
    Array [
      Object {
        "id": "default",
        "isActive": false,
        "label": "Default",
      },
      Object {
        "id": "aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b",
        "isActive": false,
        "label": "Built-in Audio Analog Stereo",
      },
      Object {
        "id": "6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb",
        "isActive": false,
        "label": "Plantronics C320 Analog Stereo",
      },
      Object {
        "id": "807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957",
        "isActive": true,
        "label": "USG PnIXAu20o  Analog Stereo",
      },
    ]
  }
  ringtoneDevices={
    Array [
      Object {
        "id": "default",
        "isActive": true,
        "label": "Default",
      },
      Object {
        "id": "aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b",
        "isActive": true,
        "label": "Built-in Audio Analog Stereo",
      },
      Object {
        "id": "6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb",
        "isActive": false,
        "label": "Plantronics C320 Analog Stereo",
      },
      Object {
        "id": "807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957",
        "isActive": true,
        "label": "USG PnIXAu20o  Analog Stereo",
      },
    ]
  }
  speakerDevices={
    Array [
      Object {
        "id": "default",
        "isActive": false,
        "label": "Default",
      },
      Object {
        "id": "aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b",
        "isActive": true,
        "label": "Built-in Audio Analog Stereo",
      },
      Object {
        "id": "6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb",
        "isActive": true,
        "label": "Plantronics C320 Analog Stereo",
      },
      Object {
        "id": "807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957",
        "isActive": false,
        "label": "USG PnIXAu20o  Analog Stereo",
      },
    ]
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
  updateActiveOutputNotificationDevice={[Function]}
  updateActiveOutputRingtoneDevice={[Function]}
  updateActiveOutputSpeakerDevice={[Function]}
/>
`;

exports[`<AudioOutputMenu /> renders the audio output menu connected with redux when notifications are not available in zendesk mode 1`] = `
<AudioOutputMenu
  audioNotificationsEnabled={false}
  crmModule="zendesk"
  notificationDevices={
    Array [
      Object {
        "id": "default",
        "isActive": false,
        "label": "Default",
      },
      Object {
        "id": "aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b",
        "isActive": false,
        "label": "Built-in Audio Analog Stereo",
      },
      Object {
        "id": "6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb",
        "isActive": false,
        "label": "Plantronics C320 Analog Stereo",
      },
      Object {
        "id": "807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957",
        "isActive": true,
        "label": "USG PnIXAu20o  Analog Stereo",
      },
    ]
  }
  ringtoneDevices={
    Array [
      Object {
        "id": "default",
        "isActive": true,
        "label": "Default",
      },
      Object {
        "id": "aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b",
        "isActive": true,
        "label": "Built-in Audio Analog Stereo",
      },
      Object {
        "id": "6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb",
        "isActive": false,
        "label": "Plantronics C320 Analog Stereo",
      },
      Object {
        "id": "807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957",
        "isActive": true,
        "label": "USG PnIXAu20o  Analog Stereo",
      },
    ]
  }
  speakerDevices={
    Array [
      Object {
        "id": "default",
        "isActive": false,
        "label": "Default",
      },
      Object {
        "id": "aead4bb457d9b0aeeddb669d8e6fef68d9cf2379d25264d6948af1dbdea3259b",
        "isActive": true,
        "label": "Built-in Audio Analog Stereo",
      },
      Object {
        "id": "6d4096a33f2ba164d976a9929c3936b2956644e0581085c8870c4272782db9fb",
        "isActive": true,
        "label": "Plantronics C320 Analog Stereo",
      },
      Object {
        "id": "807fbf9ab5fd57290d125721e10edaab372b2c52569513de20934e0a14971957",
        "isActive": false,
        "label": "USG PnIXAu20o  Analog Stereo",
      },
    ]
  }
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "unsubscribe": [Function],
    }
  }
  updateActiveOutputNotificationDevice={[Function]}
  updateActiveOutputRingtoneDevice={[Function]}
  updateActiveOutputSpeakerDevice={[Function]}
/>
`;

exports[`<AudioOutputMenu /> renders the audio output menu when crmModule is not zendesk and notifications are not enabled in toolbar mode 1`] = `
<React.Fragment>
  <DeviceMenu
    audio="ringtone"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Ringtone",
        "id": "app.containers.AudioOutputMenu.ringtone",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
  <DeviceMenu
    audio="voice"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Voice",
        "id": "app.containers.AudioOutputMenu.voice",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
</React.Fragment>
`;

exports[`<AudioOutputMenu /> renders the audio output menu when crmModule is zendesk in desktop mode 1`] = `
<React.Fragment>
  <DeviceMenu
    audio="media"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Media",
        "id": "app.containers.AudioOutputMenu.media",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
  <DeviceMenu
    audio="ringtone"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Ringtone",
        "id": "app.containers.AudioOutputMenu.ringtone",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
  <DeviceMenu
    audio="voice"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Voice",
        "id": "app.containers.AudioOutputMenu.voice",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
</React.Fragment>
`;

exports[`<AudioOutputMenu /> renders the audio output menu when crmModule is zendesk in toolbar mode 1`] = `
<React.Fragment>
  <DeviceMenu
    audio="media"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Media",
        "id": "app.containers.AudioOutputMenu.media",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
  <DeviceMenu
    audio="ringtone"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Ringtone",
        "id": "app.containers.AudioOutputMenu.ringtone",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
  <DeviceMenu
    audio="voice"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Voice",
        "id": "app.containers.AudioOutputMenu.voice",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
</React.Fragment>
`;

exports[`<AudioOutputMenu /> renders the audio output menu when devices lists are empty 1`] = `
<React.Fragment>
  <DeviceMenu
    audio="media"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Media",
        "id": "app.containers.AudioOutputMenu.media",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
  <DeviceMenu
    audio="ringtone"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Ringtone",
        "id": "app.containers.AudioOutputMenu.ringtone",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
  <DeviceMenu
    audio="voice"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Voice",
        "id": "app.containers.AudioOutputMenu.voice",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
</React.Fragment>
`;

exports[`<AudioOutputMenu /> renders the audio output menu when notifications are not enabled 1`] = `
<React.Fragment>
  <DeviceMenu
    audio="media"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Media",
        "id": "app.containers.AudioOutputMenu.media",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
  <DeviceMenu
    audio="ringtone"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Ringtone",
        "id": "app.containers.AudioOutputMenu.ringtone",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
  <DeviceMenu
    audio="voice"
    devices={Array []}
    label={
      Object {
        "defaultMessage": "Voice",
        "id": "app.containers.AudioOutputMenu.voice",
      }
    }
    setDeviceAsActive={[MockFunction]}
  />
</React.Fragment>
`;
